src/main/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResource.java

```diff
@@ -11,20 +10,30 @@
 import javax.ws.rs.core.Response;
 import org.apache.commons.lang3.tuple.ImmutablePair;
 import org.broadinstitute.dsde.consent.ontology.datause.DataUseMatcher;
+import org.broadinstitute.dsde.consent.ontology.datause.DataUseMatcherV3;
+import org.broadinstitute.dsde.consent.ontology.datause.MatchResult;
+import org.broadinstitute.dsde.consent.ontology.datause.MatchResultType;
 import org.broadinstitute.dsde.consent.ontology.model.DataUse;
 import org.broadinstitute.dsde.consent.ontology.model.DataUseMatchPair;
+import org.broadinstitute.dsde.consent.ontology.model.DataUseMatchPairV3;
+import org.broadinstitute.dsde.consent.ontology.model.DataUseV3;
 
 @Path("/match")
 @Consumes("application/json")
 @Produces("application/json")
 public class MatchResource {
 
-    private final DataUseMatcher dataUseMatcher;
+    private DataUseMatcher dataUseMatcher;
+
+    private DataUseMatcherV3 dataUseMatcherV3;
 
-    @Inject
     MatchResource(DataUseMatcher dataUseMatcher) {
         this.dataUseMatcher = dataUseMatcher;
     }
+    MatchResource(DataUseMatcherV3 dataUseMatcherV3) {
+      this.dataUseMatcherV3 = dataUseMatcherV3;
+    }
```

Comment:
We should have only one constructor:
```suggestion
    MatchResource(DataUseMatcher dataUseMatcher, DataUseMatcherV3 dataUseMatcherV3) {
      this.dataUseMatcher = dataUseMatcher;
      this.dataUseMatcherV3 = dataUseMatcherV3;
    }
```